crates/cranelift/src/compiler.rs

```diff
@@ -537,47 +551,110 @@ impl Compiler {
         context.func =
             ir::Function::with_name_signature(ir::ExternalName::user(0, 0), wasm_signature);
 
-        let ss = context.func.create_stack_slot(ir::StackSlotData::new(
+        let mut builder = FunctionBuilder::new(&mut context.func, func_translator.context());
+        let block0 = builder.create_block();
+
+        let (values_vec_ptr_val, values_vec_len) =
+            self.wasm_to_host_spill_args(ty, &mut builder, block0);
+
+        let block_params = builder.func.dfg.block_params(block0);
+        let callee_args = [
+            block_params[0],
+            block_params[1],
+            values_vec_ptr_val,
+            builder
+                .ins()
+                .iconst(pointer_type, i64::from(values_vec_len)),
+        ];
+
+        let new_sig = builder.import_signature(host_signature);
+        let callee_value = builder.ins().iconst(pointer_type, host_fn as i64);
+        builder
+            .ins()
+            .call_indirect(new_sig, callee_value, &callee_args);
```

Comment:
That's a good question...

Looking into it it appears that it's because the compiler type is defined in the `wasmtime-environ` crate but the types for the function pointer signature (e.g. `VMContext` and `ValRaw`) are located in the `wasmtime-runtime` crate which `wasmtime-cranelift` doesn't depend on, so I think the dependency tree is thwarting that plumbing of information.